File: content/problems/0049-group-anagrams/solutions/sorted-key-hash/Solution.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> groups = new LinkedHashMap<>();
        for (String value : strs) {
            char[] chars = value.toCharArray();
            Arrays.sort(chars);
            String key = new String(chars);
            groups.computeIfAbsent(key, ignored -> new ArrayList<>()).add(value);
        }
        return new ArrayList<>(groups.values());
    }
}

File: content/problems/0049-group-anagrams/solutions/sorted-key-hash/Solution.kt
class Solution {
    fun groupAnagrams(strs: Array<String>): List<List<String>> {
        val groups = linkedMapOf<String, MutableList<String>>()
        for (value in strs) {
            val chars = value.toCharArray()
            chars.sort()
            val key = chars.concatToString()
            groups.getOrPut(key) { ArrayList() }.add(value)
        }
        return groups.values.map { it.toList() }
    }
}
